### DISCUSSION TITLE
求新功能

### DISCUSSION THREAD

**Discussion by zxp2019** [luosiwei-cmd/CarYon#19] [General]
在 [Wiki](https://github.com/luosiwei-cmd/CarYon/wiki/%E5%85%B6%E4%BB%96%E9%9A%8F%E6%9C%BA%E5%87%BD%E6%95%B0) 中未找到以下功能，希望能添加。如果项目中本来就有以下功能，请告知使用方法，谢谢。

- 生成长度为 n 的随机整数序列，序列中每个数均为范围在 [1,k] 之间的随机数（k>=n），且序列内的数不重复，n 和 k 由用户指定。
- 生成长度为 n 的随机整数**升序**序列，序列中每个数均为范围在 [1,k] 之间的随机数（k>=n），且序列内的数不重复，n 和 k 由用户指定。

支持你们的项目，加油！

**Comment by luosiwei-cmd** [comment]
您好，该功能可以通过用户自行利用cyrand函数存储在数组中，不断检测是否重复，否则删除的方式做成。

如果您急需此功能，您还得自行制作，如果您能等到周四上线，那么谢谢您的支持！

**Comment by luosiwei-cmd** [comment]
`因为这几天不能用电脑qwq`

**Comment by zxp2019** [comment]
感谢！

**Comment by luosiwei-cmd** [comment]
您好，功能已被实现，稍后更新。

代码如下：

```cpp
	inline bool checktmp(int *arr,int tmp,int flag,int end){
		if(flag==0){
			return true;
		}
		for(int i=0;i<end;i++){
			else if(flag==-1){
				if(arr[i]==tmp){
					return false;
				}
			}
			else if(flag==1){
				if(arr[i]>tmp){
					return false;
				}
			}
			else if(flag==2){
				if(arr[i]<tmp){
					return false;
				}
			}
		}
		return true;
	}
	inline void randomArr(int* arr,int n,int lower,int upper,int flag){
		for(int i=0;i<n;){
			int tmp=cyrand(lower,upper);
			if(checktmp(arr,tmp,flag,i-1)){
				arr[i++]=tmp;
			}
		}
	} 
```

`randomArr` 函数，第一个参数是数组，第二个是您问题中的n，然后 lower，upper是每个数的最小值和最大值。flag=0代表无要求，=-1代表不重复，=1代表升序，=2代表降序。

**Comment by zxp2019** [comment]
感谢！

**Comment by luosiwei-cmd** [comment]
您好，新功能使用如何。如果没有其他问题，9：30将关闭这个 Issue

**Comment by zxp2019** [comment]
很好，我现在就关闭了
